l1-contracts/contracts/bridge/ntv/NativeTokenVault.sol

```diff
@@ -0,0 +1,419 @@
+    function bridgeBurn(
+        uint256 _chainId,
+        uint256,
+        bytes32 _assetId,
+        address _prevMsgSender,
+        bytes calldata _data
+    ) external payable override onlyAssetRouter whenNotPaused returns (bytes memory _bridgeMintData) {
+        if (isTokenBridged[_assetId]) {
+            _bridgeMintData = _bridgeBurnBridgedToken(_chainId, _assetId, _prevMsgSender, _data);
+        } else {
+            _bridgeMintData = _bridgeBurnNativeToken({
+                _chainId: _chainId,
+                _assetId: _assetId,
+                _prevMsgSender: _prevMsgSender,
+                _depositChecked: false,
+                _data: _data
+            });
+        }
+    }
+
+    function _bridgeBurnBridgedToken(
+        uint256 _chainId,
+        bytes32 _assetId,
+        address _prevMsgSender,
+        bytes calldata _data
+    ) internal returns (bytes memory _bridgeMintData) {
+        (uint256 _amount, address _receiver) = abi.decode(_data, (uint256, address));
+        if (_amount == 0) {
+            // "Amount cannot be zero");
+            revert AmountMustBeGreaterThanZero();
+        }
+
+        address bridgedToken = tokenAddress[_assetId];
+        IBridgedStandardToken(bridgedToken).bridgeBurn(_prevMsgSender, _amount);
+
+        emit BridgeBurn({
+            chainId: _chainId,
+            assetId: _assetId,
+            sender: _prevMsgSender,
+            receiver: _receiver,
+            amount: _amount
+        });
+        _bridgeMintData = _data;
+    }
+
+    function _bridgeBurnNativeToken(
+        uint256 _chainId,
+        bytes32 _assetId,
+        address _prevMsgSender,
+        bool _depositChecked,
+        bytes calldata _data
+    ) internal virtual returns (bytes memory _bridgeMintData) {
+        (uint256 _depositAmount, address _receiver) = abi.decode(_data, (uint256, address));
+
+        uint256 amount;
+        address nativeToken = tokenAddress[_assetId];
+        if (nativeToken == ETH_TOKEN_ADDRESS) {
+            amount = msg.value;
+
+            // In the old SDK/contracts the user had to always provide `0` as the deposit amount for ETH token, while
+            // ultimately the provided `msg.value` was used as the deposit amount. This check is needed for backwards compatibility.
+            if (_depositAmount == 0) {
+                _depositAmount = amount;
+            }
+
+            if (_depositAmount != amount) {
+                revert ValueMismatch(amount, msg.value);
+            }
+        } else {
+            // The Bridgehub also checks this, but we want to be sure
+            if (msg.value != 0) {
+                revert NonEmptyMsgValue();
+            }
+
+            amount = _depositAmount;
+            if (!_depositChecked) {
+                uint256 expectedDepositAmount = _depositFunds(_prevMsgSender, IERC20(nativeToken), _depositAmount); // note if _prevMsgSender is this contract, this will return 0. This does not happen.
```

Comment:
If it can't perform baseToken deposits then all L1->L2 txs will fail, which would be ver bad..? Maybe you meant L2->L1 txs?

Reply:
You mean the: 
`// note if _prevMsgSender is this contract, this will return 0. This does not happen.
`line? I don't think it applies, since the _prevMsg sender is the user

Reply:
I meant if NTV itself will ever try to perform an L1->L2 tx ("direct" one, not two bridges), the deposit wont work since  _prevMsgSender is this contract